sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -1617,12 +1623,20 @@ void CAFMaker::produce(art::Event& evt) noexcept {
     const recob::PFParticle *primary = (iPart == fmPFPart.size()) ? NULL : fmPFPart[iPart].get();
     const larpandoraobj::PFParticleMetadata *primary_meta = (iPart == fmPFPart.size()) ? NULL : fmPFPMeta.at(iPart).at(0).get();
     // get the flash match
-    const sbn::SimpleFlashMatch* fmatch = nullptr;
-    if (fm_sFM.isValid() && primary != NULL) {
-      std::vector<art::Ptr<sbn::SimpleFlashMatch>> fmatches = fm_sFM.at(iPart);
-      if (fmatches.size() != 0) {
-        assert(fmatches.size() == 1);
-        fmatch = fmatches[0].get();
+
+    std::map<std::string, const sbn::SimpleFlashMatch*> fmatch_map;
+    std::map<std::string, art::FindManyP<sbn::SimpleFlashMatch>>::iterator fmatch_it;
+    for(fmatch_it = fmatch_assn_map.begin();fmatch_it != fmatch_assn_map.end();fmatch_it++) {
+      auto fname = fmatch_it->first;
+      auto fm_sFM = fmatch_it->second;
+      const sbn::SimpleFlashMatch* fmatch = nullptr;
+      if (fm_sFM.isValid() && primary != NULL) {
```

Comment:
Can you substitute this `NULL` for a `nullptr`? 

If it needs to be updated in several places then don't worry.